tests/unit/test_lib_time.py

```diff
@@ -102,6 +154,45 @@ def test___datetime_with_microseconds___convert_to_timestamp___is_reversible(
     assert roundtrip_dt_microsecond == microsecond
 
 
+@pytest.mark.parametrize(
+    "base_dt, microsecond, subseconds",
+    [
+        (JAN_01_1850_DATETIME, 0, 0),
+        (JAN_01_1850_DATETIME, 1, 0x10C6F7A0B5EE),
+        (JAN_01_1850_DATETIME, 250000, 0x4000000000000000),
+        (JAN_01_1850_DATETIME, 500000, 0x8000000000000000),
```

Comment:
These subseconds values seem to be based on the premise that the NI-BTF fractional seconds always count forwards (like Protobuf timestamps), even when the whole seconds are negative. Unfortunately, https://www.ni.com/docs/en-US/bundle/labwindows-cvi/page/cvi/libref/ni-btf.htm doesn't say whether the positive fractions of a second follow the sign of the whole seconds.

Do we have any documentation or tests confirming that this is the case?

@zhindes

Reply:
BTW, I am 90% sure that for CVI time, lsb implicitly follows the sign of msb. Why? Because these types are defined as 128-bit fixed point numbers with 64-bit int part and 64-bit fractional part.

If we don't have any docs/tests confirming this, it would be helpful to add a test case for this to the internal "nitypest" component.

Reply:
It would be pretty insane for a generic fixed point type to not have the sign behavior you describe...

Reply:
I'm just going off from what @zhindes documented in AbsoluteTime which says that lsb is `positive fractions (2^-64) of a second`, and is also mentioned in the NI_BTF documentation
![image](https://github.com/ni/nidaqmx-python/assets/74756143/1209b022-a2ae-4a16-b84a-5ed2396b5d3a)

which I assume is referred to from here
https://dev.azure.com/ni/DevCentral/_git/ni-central?path=/src/daqmx/api/nicaistub/source/nicaistub/nicaistub_typedefs.h&version=GCde6458e0b93bf6d6c064623bb74cef4ab9f5cc84&line=80&lineEnd=84&lineStartColumn=1&lineEndColumn=11&lineStyle=plain&_a=contents
I also noticed that cvi has it as an [unsigned int](https://dev.azure.com/ni/DevCentral/_git/ni-central?path=/src/cvi_product/cvi/Components/cvi_Headers/source/cvitdms.h&version=GCde6458e0b93bf6d6c064623bb74cef4ab9f5cc84&line=46&lineEnd=46&lineStartColumn=26&lineEndColumn=42&lineStyle=plain&_a=contents) so maybe this proves that it's always positive?

Reply:
The lsb is always positive because the sign bit is in the msb. This does not tell you whether the lsb counts forwards or backwards in time.

I wrote some new nitypest test cases and convinced myself that the lsb always counts forwards in time because it's the lower half of a 128-bit two's complement number. It's like how adding 1 to 0xFFFFFFF0 (-16) yields 0xFFFFFFF1 (-15), which is less negative.

Sorry about the confusion.